proxy/proxy.go

```diff
@@ -46,15 +46,15 @@ type Proxy interface {
 	toMap() map[string]interface{}
 }
 
-func NewProxy(protocol string, u *url.URL, src string) (Proxy, error) {
+func NewProxy(u *url.URL, src string) (Proxy, error) {
 	proxy := new(proxy)
-	if err := proxy.init(protocol, u, src); err != nil {
+	if err := proxy.init(u, src); err != nil {
 		return nil, err
 	}
 	return proxy, nil
 }
 
-var defaultPorts = map[string]uint16{"": 8080, "https": 8443}
+var defaultPorts = map[string]uint16{"": 8443, "http": 8443, "https": 8443}
```

Comment:
They are defaults as were previously specified by the Insight Agent. Keeping 8443 so as to be consistent with that implementation.
As far as I know there is no RFC default port, but `8080` is the standard used by many client implementations, so that is a deviation here. `8080` would be the ideal value for these.